app/models/concerns/eds_subjects.rb

```diff
@@ -0,0 +1,99 @@
+module EdsSubjects
+  def eds_subjects
+    @eds_subjects ||= Subject.from(self['eds_subjects'])
+  end
+
+  def eds_subjects_geographic
+    @eds_subjects_geographic ||= Subject.from(self['eds_subjects_geographic'])
+  end
+
+  def eds_subjects_person
+    @eds_subjects_person ||= Subject.from(self['eds_subjects_person'])
+  end
+
+  def eds_author_supplied_keywords
+    @eds_author_supplied_keywords ||= Subject.from(self['eds_author_supplied_keywords'])
+  end
+
+  # EDS-centric subject with linking capability
+  class Subject
+    attr_reader :terms
+    delegate :present?, to: :terms
+
+    class Term < Struct.new(:fieldCode, :searchTerm, :label)
+      delegate :to_s, to: :label
+
+      def to_html
+        field = case fieldCode
+                when 'SH'
+                  'subject_heading'
+                when 'DE'
+                  'descriptor'
+                when 'SU'
+                  'subject'
+                when 'KW'
+                  'keyword'
+                else
+                  'subject'
+                end
+        # TODO: we don't have a view context for `link_to`
+        path = Rails.application.routes.url_helpers.articles_path
+        "<a href=\"#{path}?q=#{CGI.escape(searchTerm)}&amp;search_field=#{field}\">#{CGI.escapeHTML(label)}</a>"
```

Comment:
We could pass `q` and `search_field` arguments to `articles_path` instead of constructing the url w/ `?` and `&amp;` manually.

Reply:
oh right 👍